editor/src/components/inspector/inspector-common.ts

```diff
@@ -1120,12 +1126,54 @@ export function toggleAbsolutePositioningCommands(
           : []),
       ]
     } else {
-      return [
-        ...sizeToVisualDimensions(jsxMetadata, elementPathTree, elementPath),
-        ...addPositionAbsoluteTopLeft(jsxMetadata, elementPath),
-      ]
+      return getConvertIndividualElementToAbsoluteCommandsFromMetadata(
+        elementPath,
+        jsxMetadata,
+        elementPathTree,
+      )
     }
   })
 
   return commands
 }
+
+export function getConvertIndividualElementToAbsoluteCommandsFromMetadata(
+  target: ElementPath,
+  jsxMetadata: ElementInstanceMetadataMap,
+  elementPathTree: ElementPathTrees,
+): Array<CanvasCommand> {
+  const localFrame = MetadataUtils.getFrame(target, jsxMetadata)
+  if (localFrame == null || isInfinityRectangle(localFrame)) {
+    return []
+  }
+
+  const element = MetadataUtils.findElementByElementPath(jsxMetadata, target)
+  if (element == null) {
+    return []
+  }
+
+  const parentFlexDirection = element.specialSizeMeasurements.parentFlexDirection
+
+  return getConvertIndividualElementToAbsoluteCommands(
+    target,
+    jsxMetadata,
+    elementPathTree,
+    localFrame,
+    parentFlexDirection,
+  )
+}
+
+export function getConvertIndividualElementToAbsoluteCommands(
+  target: ElementPath,
+  jsxMetadata: ElementInstanceMetadataMap,
+  elementPathTree: ElementPathTrees,
+  frame: LocalRectangle,
+  parentFlexDirection: FlexDirection | null,
+): Array<CanvasCommand> {
+  // First round the frame so that we don't end up with half pixel values
+  const roundedFrame = roundRectangleToNearestWhole(frame)
+  return [
+    ...sizeToDimensionsFromFrame(jsxMetadata, elementPathTree, target, roundedFrame),
```

Comment:
I have a feeling that this is the point where you could clear layout props like right,bottom,flex-basis,min-width,max-width etc that could interfere with the visual layout – this is the crossover with @bkrmendy 's PR which is removing the interfering props

Reply:
Yeah, that's why I commented on that PR asking that it not be merged until after this one, as that's what is already happening as part of `sizeToDimensionsFromFrame`. But it's not removing `right` and `bottom`. I'll include those 2 props in the pruning (this is also why I removed those initial layout props from a couple of tests, as those tests were providing all of the TLBRWH props!)

Reply:
> as those tests were providing all of the TLBRWH props

and since the user _can_ write crazy code like that, I'd actually add those back to the initial conditions, and make sure the final result only has TLWH

Reply:
Yep, that's fair. I've put them back now and included the BR pin pruning